packages/vscode-extension/lib/metro_helpers.js

```diff
@@ -128,10 +128,26 @@ function adaptMetroConfig(config) {
   // that adds a preamble to one of the files loaded by React Native in initialization.
   // It also provides a way to load some integration files when apropriate libraries are requested.
   // Since the transformer is loaded by path, we pass the original transformer that it wraps via process.env
-  process.env.REACT_NATIVE_IDE_ORIG_BABEL_TRANSFORMER_PATH =
-    config.transformer.babelTransformerPath;
+  process.env.RADON_IDE_ORIG_BABEL_TRANSFORMER_PATH = config.transformer.babelTransformerPath;
   config.transformer.babelTransformerPath = path.join(extensionLib, "./babel_transformer.js");
 
+  // In development, metro may resolve dependencies for the extension lib file to the extension's node_modules
+  // folder, as it lies on the path up the directory tree.
+  // Since we don't want this, we use resolver's blockList to exclude the extension lib's node_modules folder from
+  // being considered:
+  const extensionLibNodeModules = path.resolve(path.join(extensionLib, "../node_modules"));
+  let origBlockList = [];
+  if (config.resolver.blockList) {
+    // if block list is array, we use it as original block list
+    if (Array.isArray(config.resolver.blockList)) {
+      origBlockList = config.resolver.blockList;
+    } else {
+      // otherwise we create a new array containing the original block list
+      origBlockList = [config.resolver.blockList];
+    }
+  }
+  config.resolver.blockList = [new RegExp(extensionLibNodeModules + "/.*"), ...origBlockList];
```

Comment:
Are you sure this works for Windows? I mean, if `RegExp` properly handles backslashes in `extensionLibNodeModules`, especially that we potentially mix it with a forward slash

Reply:
This is actually only needed in extension development process. When extension is installed from the store, it doesn't have node_modules folder under the extension lib location. I maybe able to fix this by adding ".*" as path element when joining.